agent/src/main/java/com/teamscale/jacoco/agent/options/AgentOptions.java

```diff
@@ -203,6 +211,24 @@ public String getOriginalOptionsString() {
 		return originalOptionsString;
 	}
 
+	/**
+	 * Remove parts of the API key for security reasons from the options string.
+	 * String is used for logging purposes.
+	 */
+	public String getObfuscatedOptionsString() {
+		// Example for `getOriginalOptionsString()`:
+		// 		config-file=jacocoagent.properties,teamscale-access-token=unlYgehaYYYhbPAegNWV3WgjOzxkmNHn
+		Pattern pattern = Pattern.compile("(.*-access-token=)([^,]+)(.*)");
```

Comment:
I might not know enough, but are you sure there's no easier way to read and write the value of the access token than parsing the entire options string with a regex? I'd at least double check there are no better ways in the agent code already.